murdock/job.py

```diff
@@ -200,6 +210,14 @@ def __eq__(self, other) -> bool:
     def __hash__(self) -> int:
         return hash(self.uid)
 
+    def set_start_time(self, start_time: datetime):
+        assert start_time.tzinfo is not None
```

Comment:
I put these here specifically so that they are triggered if we ever attempt to abuse the internal api and to ensure that it shows up during tests. But I can remove them if you prefer.

> Maybe just return if tzinfo is None ?

I think this would lead to unpredictable behavior. I'd rather throw an exception in that case then.

Reply:
> I think this would lead to unpredictable behavior. I'd rather throw an exception in that case then.

Seems like a good solution. Let's go for it.